app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -196,5 +201,22 @@ def breadcrumbs_options
     def x_node_right_cell
       @sb[@sb[:active_accord]].presence || x_node
     end
+
+    def breadcrumb_get_session
+      @x_node_text = session[:x_node_text]
+    end
+
+    def breadcrumb_set_session
+      session[:x_node_text] = @x_node_text
+    end
+
+    def x_node_text
+      @x_node_text[x_active_tree] if @x_node_text
+    end
+
+    def x_node_text_set
```

Comment:
What about having this parameterized and pushing the `params[:text]` into it where it is being called? Then you can rename it to `x_node_text=`. Ruby is awesome, you can have `=` methods with a single argument and then you can call ` x_node_text = params[:text]`.